src/IO/Observer/TemporalId.hpp

```diff
@@ -0,0 +1,73 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <functional>
+#include <string>
+
+#include "Utilities/PrettyType.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace observers {
+/*!
+ * \ingroup ObserversGroup
+ * \brief A type-erased temporal identifier that combines the identifier's type
+ * and hash.
+ *
+ * The identifier must also have a `value()` method that returns a double
+ * representing the current "time".
+ *
+ * A specialization of `std::hash` is provided to allow using `TemporalId`
+ * as a key in associative containers.
+ */
+class TemporalId {
```

Comment:
What is this class for?  Why not just use a double (or simple wrapper around same)?

I don't think we want observations associated with a TemporalId (in the evolution sense) anyway, but just a (little 't') time (double).